src/Nethermind/Nethermind.Merge.AuRa/AuRaPostMergeBlockProducer.cs

```diff
@@ -44,12 +44,10 @@ public override Block PrepareEmptyBlock(BlockHeader parent, PayloadAttributes? p
         {
             var block = base.PrepareEmptyBlock(parent, payloadAttributes);
 
-            // processing is only done to apply AuRa block rewards and rewrite posdao contracts
-            // TODO should we return null instead of empty block if we couldn't process?
             if (TrySetState(parent.StateRoot))
-                block = ProcessPreparedBlock(block, null) ?? block;
+                return ProcessPreparedBlock(block, null) ?? throw new System.Exception("Couldn't process empty block");
 
-            return block;
+            throw new System.Exception("Couldn't process empty block");
```

Comment:
Don't ever throw `System.Exception`, throw something meaningful

Reply:
Yeah, I was hoping that the exception type I'm looking for would already exist and you would point me to it